_test/test_utilities_herbert/test_fast_map.m

```diff
@@ -0,0 +1,166 @@
+classdef test_fast_map < TestCase
+    properties
+    end
+    methods
+        function obj = test_fast_map(varargin)
+            if nargin<1
+                name = 'test_fast_map';
+            else
+                name = varargin{1};
+            end
+            obj = obj@TestCase(name);
+        end
+        %------------------------------------------------------------------
+        %------------------------------------------------------------------
+        function test_build_uint64_map_from_constructor(~)
+            keys= uint64([10,20,30]);
+            val  = 1:3;
+            fm = fast_map(keys,val);
+
+            assertEqual(fm.KeyType,'uint64');
+            assertEqual(fm.keys,keys);
+            assertEqual(fm.values,val);
+        end
+
+        function test_set_different_key_type(~)
+            fm = fast_map();
+            fm.KeyType = uint64(1);
+            fm = fm.add(10,1);
+
+            assertEqual(fm.KeyType,'uint64');
+            assertEqual(fm.keys,uint64(10));
+            assertEqual(fm.values,1);
+        end
+        %------------------------------------------------------------------
+        function test_get_all_val_for_keys_optimized_no_checks(~)
+            n_keys = 100;
+            base_key = 10+round(rand(1,10*n_keys)*(10*n_keys-1));
+            base_key = unique(base_key);
+            n_keys = numel(base_key);
```

Comment:
you have not said what this does, so the conversation is not resolved. Please add comments. You may refer to your benchmarking script in doing so

Reply:
"% generate specified number of random keys in the range
% exceeding  number of keys by 10. This approximate runids obtained from
% two  independent experiments provided in two cycles which are not too far
% from each other in time."
This is row abowe row 36. What else you want?